src/Avalonia.Controls/SplitButton/SplitButton.cs

```diff
@@ -0,0 +1,471 @@
+                if (_isAttachedToLogicalTree)
+                {
+                    // Must unregister events here while a reference to the old command still exists
+                    if (e.OldValue.GetValueOrDefault() is ICommand oldCommand)
+                    {
+                        oldCommand.CanExecuteChanged -= CanExecuteChanged;
+                    }
+
+                    if (e.NewValue.GetValueOrDefault() is ICommand newCommand)
+                    {
+                        newCommand.CanExecuteChanged += CanExecuteChanged;
+                    }
+                }
+
+                CanExecuteChanged(this, EventArgs.Empty);
+            }
+            else if (e.Property == CommandParameterProperty)
+            {
+                CanExecuteChanged(this, EventArgs.Empty);
+            }
+            else if (e.Property == FlyoutProperty)
+            {
+                var oldFlyout = e.OldValue.GetValueOrDefault() as FlyoutBase;
+                var newFlyout = e.NewValue.GetValueOrDefault() as FlyoutBase;
+
+                // If flyout is changed while one is already open, make sure we 
+                // close the old one first
+                // This is the same behavior as Button
+                if (oldFlyout != null &&
+                    oldFlyout.IsOpen)
+                {
+                    oldFlyout.Hide();
+                }
+
+                // Must unregister events here while a reference to the old flyout still exists
+                if (oldFlyout != null)
+                {
+                    UnregisterFlyoutEvents(oldFlyout);
+                }
+
+                if (newFlyout != null)
+                {
+                    RegisterFlyoutEvents(newFlyout);
+                }
+
+                UpdatePseudoClasses();
+            }
+
+            base.OnPropertyChanged(e);
+        }
+
+        /// <inheritdoc/>
+        protected override void OnKeyDown(KeyEventArgs e)
+        {
+            var key = e.Key;
+
+            if (key == Key.Space || key == Key.Enter) // Key.GamepadA is not currently supported
+            {
+                UpdatePseudoClasses();
+            }
+
+            base.OnKeyDown(e);
+        }
+
+        /// <inheritdoc/>
+        protected override void OnKeyUp(KeyEventArgs e)
+        {
+            var key = e.Key;
+
+            if (key == Key.Space || key == Key.Enter) // Key.GamepadA is not currently supported
+            {
+                UpdatePseudoClasses();
+
+                // Consider this a click on the primary button
+                if (IsEffectivelyEnabled)
+                {
+                    OnClickPrimary(null);
+                    e.Handled = true;
+                }
+            }
+            else if (key == Key.Down && IsEffectivelyEnabled)
+            {
+                // WinUI requires the VirtualKey.Menu (alt) + VirtualKey.Down to open the flyout
+                // Avalonia will only require Key.Down which is better cross-platform
```

Comment:
Comment was made: https://github.com/AvaloniaUI/Avalonia/pull/7422#issuecomment-1021911616

 > "Down" should not be used to open flyout. We might have XY navigation in the future (see how UWP works with arrow buttons navigation).

Alright, I will add back the alt-key requirement if I can find it in the mapping. There is no VirtualKey.Menu in Avalonia as far as I know.

Reply:
```
if (key == Key.Down && e.KeyModifiers.HasAllFlags(KeyModifiers.Alt) && IsEffectivelyEnabled)
```
works just fine for me

Reply:
Updated and resolved.